### DISCUSSION TITLE
Web Runner For Editor

### DISCUSSION THREAD

**Discussion by PaulHMason** [godotengine/godot-proposals#11157] [Editor]
I use Godot for web games and have found that the workflow for web games that require significant JavaScript interop is very cumbersome.

**Background**
I have a game that is embedded in an HTML page that communicates with a 3rd party JS library, that in turn communicates with a real-time service. It also handles various UI overlays.

**Problem**
The dev runner in Godot is a native application, so I have to mock out the server comms and provide some fairly complex request and response packets during development. This is very time consuming and there are certain things I just can't reproduce (such as the html overlays). It would be far easier if I could just use these dependencies directly during dev/test instead of providing an incomplete mocked solution.

**Proposal**
I think it would be much easier to develop web games if Godot provided an optional "web runner", where the game can be run from the editor in a web view (with the usual debugging capabilities). Developers can then provide their own custom html template page that is embedded in the "web runner" that includes whatever dependencies they require. That way you will have a seamless dev/test cycle using the game in the manner it will be run in production.

Thoughts on the viability of this?

**Comment by ShalokShalom** [comment]


**Reply by Zireael07** [reply]
That's not communication (you can already communicate from HTML5 to Godot via the Javascript bridge), that's a scripting language implementation, i.e. a GDScript replacement

**Reply by ShalokShalom** [reply]
Yes, I thought you could possibly usw that as communication tool.

Thanks for clarifying that to me.
